src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMIT.java

```diff
@@ -35,37 +43,59 @@ public void tearDown() throws IOException, InterruptedException {
         project.delete();
     }
 
-    @Ignore("TODO: Fix this test")
     @Test
     public void testCheckout() throws Exception {
-        project.setScm(createSCM("**/master"));
+        project.setScm(createSCM("*/master"));
+        project.save();
 
         FreeStyleBuild build = project.scheduleBuild2(0).get();
 
         assertEquals(Result.SUCCESS, build.getResult());
         assertTrue(build.getWorkspace().child("add_file").isDirectory());
     }
 
+    @Test
+    public void testCheckoutWithMultipleBranches() throws Exception {
+        project.setScm(createSCM("*/master", "*/basic_branching"));
+        project.save();
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        assertEquals(Result.SUCCESS, build.getResult());
+        assertTrue(build.getWorkspace().child("add_file").isDirectory());
+
+        waitFor(() -> {
+            if (project.isInQueue()) {
+                return Optional.of("Build is queued");
+            }
+            return Optional.empty();
+        }, 1000);
+        assertEquals(2, Iterables.size(project.getBuilds()));
```

Comment:
This is convenient, maybe we could just use the `size()` method given it is a test? After all, the only thing the `Iterables.size()` does right now (as things stand) is to call the `size()` method anyway.

Reply:
Unfortunately `size()` is deprecated

Reply:
I know, but given it will be invoked anyway and it is a test I think it is better, as we can then avoid adding a decade old guava as a direct dependency.

Reply:
Cool, I don't have strong feeling either way. If you think using deprecated api is better than using an older version of a library.